### ISSUE TITLE
improve retain_grad to have same functionality as tin torch 

### ISSUE THREAD

**Issue by aanurraj** [OpenMined/SyMPC#240]
## Description
Currently, we have an issue if we try to do an operation on two mpc tensors one with `requires grad = True` and the other with `requires grad = False`, and run backward to calculate the gradient.

result  of the two of the above-mentioned tensors gives grad which is not the case if we do a similar operation on the torch.

## Are you interested in working on this improvement yourself?
- Yes, I am.

## Additional Context
[] complete the mentioned task.
[] complete the todo to add the test for the same at `test_backward_with_one_requires_grad` in `mpc_tensor_tests.py`
